Adding the scan for Graphql Introspection Query Enabled

j3ssie opened the pull request:
### BCheck Contributions

* [x] BCheck compiles and executes as expected
* [x] BCheck contains appropriate metadata (name, version, author, description and appropriate tags)
* [x] Only .bcheck files have been added or modified
* [x] BCheck is in the appropriate folder
* [x] PR contains single or limited number of BChecks (Multiple PRs are preferred)
* [x] BCheck attempts to minimize false positives

PortSwiggerWiener commented on other/files/svn-exposed.bcheck:
```diff
@@ -0,0 +1,32 @@
+    tags: "exposure", "svn", "config", "file"
+
+run for each:
+    potential_path = ".svn/entries", ".svn/text", ".svn/all-wcprops", "CVS/Entries"
```
Should the CVS/Entries be in a separate CVS BCheck?

PortSwiggerWiener commented on other/files/svn-exposed.bcheck:
```diff
@@ -0,0 +1,32 @@
+
+given request then
+    # replace the potential path with the last path
+    send request called check:
```
What's this request being used for?

j3ssie replied:
the snippet below is used to append the `.svn/entries` to the last item of the folder in the URL.
```
    # replace the potential path with the last path
    send request called check:
        method: "GET"
        replacing path: `{regex_replace({regex_replace({base.request.url}, "^.*?\/.*?\/.*?\/", "/")}, "([^/]+)$", "")}{potential_path}`
```

like when the URL is `https://example.com/api/public` . this will send the request with the URL `https://example.com/api.svn/entries`
Let me know if we can do anything better than that here @PortSwiggerWiener .

PortSwiggerWiener commented on other/files/svn-exposed.bcheck:
```diff
@@ -0,0 +1,32 @@
+        method: "GET"
+        replacing path: `{regex_replace({base.request.url}, "^.*", "")}/{potential_path}`
+
+    if {latest.response.status_code} is "200" then
```
Would it be clearer to reference check1 here rather than latest?

j3ssie replied:
I use the `{latest.response}` is for the purpose above https://github.com/PortSwigger/BChecks/pull/175#discussion_r1481863110 to check for both of the requests. Do you think it would be better to do 2 separate if statements for `check` and `check1` ?

j3ssie commented on other/files/ds-store-exposed.bcheck:
```diff
@@ -0,0 +1,32 @@
+    author: "j3ssie"
+    tags: "exposure", "ds_store", "config", "file"
+
+run for each:
```
Hmm not sure what you mean here @PortSwiggerWiener, I see the author and tags are in the metadata part.

j3ssie commented on other/files/ds-store-exposed.bcheck:
```diff
@@ -0,0 +1,32 @@
+run for each:
+    potential_path = ".DS_Store"
+
+given request then
```
sure @PortSwiggerWiener, I can update that to `given path then` and also `v2-beta`

j3ssie commented:
I believe I've addressed all the issues you mentioned, @PortSwiggerWiener. Would you prefer a new PR (with this one close), or should I continue updating within this commit?

PortSwiggerWiener reviewed (approved):
Many thanks for considering the comments.

Looks good 👍

Hannah-PortSwigger commented:
@j3ssie We've made some small adjustments to your BChecks. Could you please check that these line up with your intended behavior?

j3ssie commented:
Everything appears to be in order from my end, @Hannah-PortSwigger. Please feel free to review and approve the PR for merging at your convenience

Hannah-PortSwigger commented:
Thank you very much for your contributions to the BChecks repository 😄
